File: client-app/src/models/table.ts
export interface TableDto {
  id: string;
  name: string;
  capacity: number;
  status: TableStatus;
  location: string;
}

export enum TableStatus {
  Available = 0,
  InUse = 1,
  Reserved = 2
}

export interface OpenTableResponse {
  orderId: string;
  sessionToken: string;
  tableName: string;
}

File: src/ShopManagement.Domain/Entities/Table.cs
using ShopManagement.Domain.Common;
using ShopManagement.Domain.Enums;

namespace ShopManagement.Domain.Entities;

public class Table : BaseEntity
{
    public Guid ShopId { get; set; }
    public string Name { get; set; } = string.Empty;
    public int Capacity { get; set; }
    public TableStatus Status { get; set; } = TableStatus.Available;
    public string Location { get; set; } = string.Empty;

    // Navigation properties
    public virtual Shop Shop { get; set; } = null!;
}
